src/time_integration/paired_explicit_runge_kutta/methods_PERK2.jl

```diff
@@ -0,0 +1,402 @@
+
+    return PairedExplicitRK2(num_stages, a_matrix, c, b1, bS, c_end)
+end
+
+# Constructor that calculates the coefficients with polynomial optimizer from a
+# semidiscretization
+function PairedExplicitRK2(num_stages, tspan, semi::AbstractSemidiscretization;
+                           verbose = false,
+                           bS = 1.0, c_end = 0.5)
+    eig_vals = eigvals(jacobian_ad_forward(semi))
+
+    a_matrix, c = compute_PairedExplicitRK2_butcher_tableau(num_stages,
+                                                            eig_vals, tspan,
+                                                            bS, c_end;
+                                                            verbose)
+
+    b1 = 1 - bS
+
+    return PairedExplicitRK2(num_stages, a_matrix, c, b1, bS, c_end)
+end
+
+# Constructor that calculates the coefficients with polynomial optimizer from a
+# list of eigenvalues
+function PairedExplicitRK2(num_stages, tspan, eig_vals::Vector{ComplexF64};
+                           verbose = false,
+                           bS = 1.0, c_end = 0.5)
+    a_matrix, c = compute_PairedExplicitRK2_butcher_tableau(num_stages,
+                                                            eig_vals, tspan,
+                                                            bS, c_end;
+                                                            verbose)
+
+    b1 = 1 - bS
+
+    return PairedExplicitRK2(num_stages, a_matrix, c, b1, bS, c_end)
+end
+
+# This struct is needed to fake https://github.com/SciML/OrdinaryDiffEq.jl/blob/0c2048a502101647ac35faabd80da8a5645beac7/src/integrators/type.jl#L1
+mutable struct PairedExplicitRKOptions{Callback}
+    callback::Callback # callbacks; used in Trixi
+    adaptive::Bool # whether the algorithm is adaptive; ignored
+    dtmax::Float64 # ignored
+    maxiters::Int # maximal number of time steps
+    tstops::Vector{Float64} # tstops from https://diffeq.sciml.ai/v6.8/basics/common_solver_opts/#Output-Control-1; ignored
+end
+
+function PairedExplicitRKOptions(callback, tspan; maxiters = typemax(Int), kwargs...)
+    PairedExplicitRKOptions{typeof(callback)}(callback, false, Inf, maxiters,
+                                              [last(tspan)])
+end
+
+abstract type PairedExplicitRK end
+abstract type AbstractPairedExplicitRKSingleIntegrator <: PairedExplicitRK end
+
+# This struct is needed to fake https://github.com/SciML/OrdinaryDiffEq.jl/blob/0c2048a502101647ac35faabd80da8a5645beac7/src/integrators/type.jl#L77
+# This implements the interface components described at
+# https://diffeq.sciml.ai/v6.8/basics/integrator/#Handing-Integrators-1
+# which are used in Trixi.
+mutable struct PairedExplicitRK2Integrator{RealT <: Real, uType, Params, Sol, F, Alg,
+                                           PairedExplicitRKOptions} <:
+               AbstractPairedExplicitRKSingleIntegrator
+    u::uType
+    du::uType
+    u_tmp::uType
+    t::RealT
+    dt::RealT # current time step
+    dtcache::RealT # ignored
```

Comment:
@bennibolm recently updated some SSP methods so that they work in more cases. Please have a look at his PR

Reply:
Given the duration of this review process, I suggest to keep it as it is here, create an issue for this, and tackle it in a subsequent PR.

Reply:
Fine with me

Reply:
Me or @warisa-r can tackle this together with #1886 in some time

Reply:
@DanielDoehring  [#1886](https://github.com/trixi-framework/Trixi.jl/issues/1886) has been sort of dealt with in the closed [#1871](https://github.com/trixi-framework/Trixi.jl/pull/1871) If you don't deem this too much for the current PR, I can also bring those changes here.

Reply:
Hm, lets keep it as is for the moment.

Reply:
Apologies for the confusion but changes according to [#1886](https://github.com/trixi-framework/Trixi.jl/issues/1886) have already been included in this [commit](https://github.com/trixi-framework/Trixi.jl/pull/1908/commits/0d410e4a9ca4f5c610be82c1737c4cafd3386a98).